### PR TITLE
Move metadata Reader to static database

### PR DESCRIPTION
**Pull request by vipangul**
#### Problem solved by the commit
- AIE Metadata reader is now moved to static database. This change is needed for AIE Relaod/Reset feature as it expects all the required AIE Metadata is parsed and updated in static database as part of updateDevice().
- This also includes a change to compile xdp_core for client.

#### Bug / issue (if any) fixed, which PR introduced the bug, how it was discovered
NA

#### What has been tested and how, request additional testing if necessary
Test Performed:
Verified on VCK190 and Phoenix.

### CODE REVIEW

File: src/runtime_src/xdp/profile/plugin/aie_trace/aie_trace_metadata.cpp [hunk 1]
```diff
@@ -71,21 +71,13 @@ namespace xdp {
...
-
-    metadataReader = aie::readAIEMetadata(data.first, data.second, aie_meta);
-
-    #endif
-
-    if (metadataReader == nullptr)
+    metadataReader = (VPDatabase::Instance()->getStaticInfo()).getAIEmetadataReader();
+    if (!metadataReader) {
+      std::stringstream msg;
+      msg << "Metadata reader reader couldn't be created for AIE trace";
```
Comment (pgschuey) [inline src/runtime_src/xdp/profile/plugin/aie_trace/aie_trace_metadata.cpp:77]: This message reads funny. I'd suggest rewording as "Metadata reader could not be created for AIE trace". We also don't have to create a stringstream here. It would be simpler to just put the string as the third argument to send().

**Review by pgschuey — CHANGES_REQUESTED** [review]

File: src/runtime_src/xdp/profile/plugin/aie_trace/aie_trace_metadata.cpp (hunk 1 above)
Comment (vipangul) [reply src/runtime_src/xdp/profile/plugin/aie_trace/aie_trace_metadata.cpp]: Thanks, pushed a commit addressing this at applicable plugins.

**Review by jvillarre — APPROVED** [review]
